src/main/java/gtPlusPlus/xmod/gregtech/api/metatileentity/implementations/base/GregtechMeta_MultiBlockBase.java

```diff
@@ -505,8 +506,7 @@ public int canBufferOutputs(ItemStack[] aItemOutputs, FluidStack[] aFluidOutputs
 
             // A map to hold the items we will be 'inputting' into the output buses. These itemstacks are actually the
             // recipe outputs.
-            ConcurrentSet<FlexiblePair<ItemStack, Integer>> aInputMap =
-                    new ConcurrentHashSet<FlexiblePair<ItemStack, Integer>>();
+            Set<FlexiblePair<ItemStack, Integer>> aInputMap = new HashSet<>();
```

Comment:
Why is it change to normal HashSet ? Later in the code it is used in normal loop and elements are removed
(https://github.com/GTNewHorizons/GTplusplus/pull/317/files#diff-155b83f6cf02853ae3ea8cf128c71abf6ed7fb3b2f7d7b43d62402209e8179c6R557)
The loop should be change to iterator loop or that should be reverted.

Reply:
Good catch. Thanks.